src/libtego/source/protocol/FileChannel.proto

```diff
@@ -36,38 +36,46 @@ package Protocol.Data.File;
 
 message Packet {
     optional FileHeader file_header = 1;
-    optional FileChunk file_chunk = 2;
-    optional FileChunkAck file_chunk_ack = 3;
-    optional FileHeaderAck file_header_ack = 4;
+    optional FileHeaderAck file_header_ack = 2;
+    optional FileHeaderResponse file_header_response = 3;
+    optional FileChunk file_chunk = 4;
+    optional FileChunkAck file_chunk_ack = 5;
+    optional FileTransferCompleteNotification file_transfer_complete_notification = 6;
 }
 
 message FileHeader {
-    optional uint32 file_id = 1;
-    optional uint64 size = 2;
-    optional string name = 3;
-    optional bytes sha3_512 = 4;
-    optional uint32 chunk_count = 5;
-    optional int32 time_delta = 6;
+    required uint32 file_id = 1;
+    required uint64 file_size = 2;
+    required string name = 3;
+    required bytes file_hash = 4;
```

Comment:
That actually makes a lot of sense. My reasoning had been, why bother making all of these verbose checks on the existence of params, when we can just mark them required and let protbuf handle the check for us. This will be a bit of effort to reimplement, but I'll try and get to it today.